pkgs/coverage/bin/test_with_coverage.dart

```diff
@@ -157,20 +149,18 @@ ${parser.usage}
     fail('--package is not a valid directory.');
   }
 
-  final packageName = (args['package-name'] as String?) ??
-      await _packageNameFromConfig(packageDir);
-  if (packageName == null) {
+  final pubspecPath = getPubspecPath(packageDir);
+  if (!File(pubspecPath).existsSync()) {
     fail(
-      "Couldn't figure out package name from --package. Make sure this is a "
-      'package directory, or try passing --package-name explicitly.',
+      "Couldn't find $pubspecPath. Make sure this command is run in a "
+      'package directory, or pass --package to explicitly set the directory.',
     );
   }
 
   return Flags(
     packageDir,
-    packageName,
     (args['out'] as String?) ?? path.join(packageDir, 'coverage'),
-    args['port'] as String,
+    args['port'] as String? ?? '0',
```

Comment:
Unrelated, but you could use `args.option()` and `args.flag()` to avoid casting.